server/accounts.go

```diff
@@ -268,6 +270,25 @@ func (a *Account) addClient(c *client) int {
 	return n
 }
 
+// Helper function to remove leaf nodes. If number of leafnodes gets large
+// this may need to be optimized out of linear search but believe number
+// of active leafnodes per account scope to be small and therefore cache friendly.
+// Lock should be held on account.
+func (a *Account) removeLeafNode(c *client) {
+	ll := len(a.lleafs)
+	for i, l := range a.lleafs {
+		if l == c {
+			a.lleafs[i] = a.lleafs[ll-1]
```

Comment:
Works, but if `ll` is 1, then `i==0` and `ll-1==0`. Also, unless you really want to set the slice to nil when removing the last, you could replace all that with:
```
if i != ll-1 {
  a.lleafs[i] = a.lleafs[ll-1]
}
a.lleafs = a.lleafs[0:ll-1]
```

Reply:
Yes I know I set it to itself if only 1 element. Yes I think setting to nil when gone is something we should do more of to allow memory to come back down.

Reply:
Yes in some cases I think we should do that.